doc/experiment.rst

```diff
@@ -12,316 +38,231 @@ Jupyter notebook, interactive python session, or Python file and use the
-to specify that the randomly selected learning rate parameter should be passed
-to the created models as a executable argument.
-
-.. code-block:: bash
-
-  import numpy as np
-  from smartsim import Experiment
-
-  exp = Experiment("Training-Run", launcher="auto")
-
-  # setup ensemble parameter space
-  learning_rate = list(np.linspace(.01, .5))
-  train_params = {"LR": learning_rate}
-
-  # define how each member should run
-  run = exp.create_run_settings(exe="python",
-                                exe_args="./train-model.py")
-
-  ensemble = exp.create_ensemble("Training-Ensemble",
-                                params=train_params,
-                                params_as_args=["LR"],
-                                run_settings=run,
-                                perm_strategy="random",
-                                n_models=4)
-  exp.start(ensemble, summary=True)
-
-
-A callable function can also be supplied for custom permutation strategies.  The
-function should take two arguments: a list of parameter names, and a list of
-lists of potential parameter values. The function should return a list of
-dictionaries that will be supplied as model parameters. The length of the list
-returned will determine how many ``Model`` instances are created.
-
-For example, the following is the built-in strategy ``all_perm``:
+``Ensemble`` will create the underlying model objects. An ensemble is created
+with the function ``Experiment.create_ensemble()`` and requires
+one of the subsequent sets of arguments upon initialization:
+1. run_settings and params
+2. run_settings and replicas
+3. batch_settings
+4. batch_settings, run_settings, and params
+5. batch_settings, run_settings, and replicas
+
+RunSettings
+^^^^^^^^^^^
+When running SmartSim on laptops and single node workstations,
+the base ``RunSettings`` object is used to parameterize jobs.
+``RunSettings`` includes a ``run_command`` parameter for local
+launches that utilize a parallel launch binary like
+``mpirun``, ``mpiexec``, and others. The ``RunSettings`` object is pass to an
+entity during stage initialization via the `run_settings` parameter.
+When creating a ``RunSettings`` object
+via the ``Experiment.create_run_settings()`` function, the appropriate ``RunSettings``
+object will be return based on what WLM you initialized the experiment with.
+See here to view the list of offered RunSettings WLM objects.
+
+BatchSettings
+^^^^^^^^^^^^^
+``BatchSettings`` is used to configure entities that should be launched
+as a batch on a WLM system. The ``BatchSettings`` object is applied to an
+entity during stage initialization via the `batch_settings` parameter.
+When creating a ``BatchSettings`` object
+via the ``Experiment.create_batch_settings()`` function, the appropriate ``BatchSettings``
+object will be return based on what WLM you initialized the experiment with.
+See here to view the list of offered RunSettings WLM objects.
```

Comment:
RunSettings -> `RunSettings`

And then please add the link (to the API, I think)